usersync/cookie.go

```diff
@@ -193,19 +193,24 @@ func (cookie *PBSCookie) SetCookieOnResponse(w http.ResponseWriter, setSiteCooki
 		currSize = len([]byte(httpCookie.String()))
 	}
 
-	uidsCookieStr := httpCookie.String()
+	var uidsCookieStr string
 	var sameSiteCookie *http.Cookie
 	if setSiteCookie {
+		httpCookie.Secure = true
+		uidsCookieStr := httpCookie.String()
```

Comment:
`uidsCookieStr` has already been declared and it doesn't need the `:=` operator. The `uidsCookieStr = httpCookie.String()` assignation happens in both the `true` and `false` branches of if statement in line 198, so maybe we can simplify:
```
196 -     var uidsCookieStr string
    +     uidsCookieStr := httpCookie.String()
197       var sameSiteCookie *http.Cookie
198       if setSiteCookie {
199           httpCookie.Secure = true
200 -         uidsCookieStr = httpCookie.String()
201           uidsCookieStr += SameSiteAttribute
202           sameSiteCookie = &http.Cookie{
203               Name:    SameSiteCookieName,
204               Value:   SameSiteCookieValue,
205               Expires: time.Now().Add(ttl),
206               Path:    "/",
207               Secure:  true,
208           }
209           sameSiteCookieStr := sameSiteCookie.String()
210           sameSiteCookieStr += SameSiteAttribute
211           w.Header().Add("Set-Cookie", sameSiteCookieStr)
212 -     } else {
213 -         uidsCookieStr = httpCookie.String()
214       }
215       w.Header().Add("Set-Cookie", uidsCookieStr)
216   }
usersync/cookie.go
```

Reply:
Good catch, that is actually a bug. If it was simply it didn't need the `:=` operator, then Go would flag that as a compiler error. What Go actually did is see that I used `uidsCookieStr :=` inside of a scope `{}`, so assumed I was declaring a new variable of the same name, that would only exist inside that code block. That is valid, but not the intended behavior.

Reply:
Wow

Reply:
Yeah, variable shadowing can be tricky in Go. Other languages ban it outright, like C#.